model/time.go

```diff
@@ -265,6 +335,11 @@ func (d Duration) String() string {
 		return "0s"
 	}
 
+	negative := ms < 0
+	if negative {
+		ms = -ms
+	}
```

Comment:
Slightly longer here, but it saves logic below:

```suggestion
	sign := ""
	if ms < 0 {
		sign = "-"
		ms *= -1
	}
```